app/controllers/appointments_controller.rb

```diff
@@ -10,9 +10,9 @@ def index
     authorize { current_admin.accessible_facilities(:manage_overdue_list).any? }
 
     @search_filters = index_params[:search_filters] || []
-    # We have to check to see this is the first page load where we want to apply default search filters. This
-    # is the case where the form is _not_ submitted and there are no search filters passed in.
-    if @search_filters.blank? && !index_params[:submitted]
+    # only apply default search filters on navigating to this page
+    # after initial page load, params will at least include a facility
+    if index_params.keys.empty?
```

Comment:
I decided to move away from the use of `params[:submitted]` because after the initial page load, the page will always append a facility to the query params. After some thought, I concluded that this route without any query params should probably behave identically to "first page load" anyhow, so it seems like the simpler solution to me.